internal/armada/server/handlers/submit_jobs.go

```diff
@@ -0,0 +1,103 @@
+package handlers
+
+import (
+	"context"
+
+	"github.com/G-Research/armada/internal/armada/permissions"
+	"github.com/G-Research/armada/internal/armada/repository"
+	"github.com/G-Research/armada/internal/common/validation"
+	"github.com/G-Research/armada/pkg/api"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+type authorizeOwnership func(ctx context.Context, queueName string) (bool, error)
+type newJobs func(context.Context, *api.JobSubmitRequest) ([]*api.Job, error)
+
+type submitJobs func(context.Context, *api.JobSubmitRequest) (*api.JobSubmitResponse, error)
+
+func (submit submitJobs) Validate(validateItem validation.JobSubmitRequestItemFn) submitJobs {
+	return func(ctx context.Context, request *api.JobSubmitRequest) (*api.JobSubmitResponse, error) {
+		if request.Queue == "" {
+			return nil, status.Errorf(codes.InvalidArgument, "queue is not specified")
+		}
+		if request.JobSetId == "" {
+			return nil, status.Errorf(codes.InvalidArgument, "job set is not specified")
+		}
+
+		for index, item := range request.JobRequestItems {
+			if err := validateItem(item); err != nil {
+				return nil, status.Errorf(codes.InvalidArgument, "Job with index %d is invalid: %s", index, err)
+			}
+		}
+
+		return submit(ctx, request)
+	}
+}
+
+func (submit submitJobs) Authorize(authorizeOwnership authorizeOwnership, authorize authorize, autocreate bool) submitJobs {
+	return func(ctx context.Context, request *api.JobSubmitRequest) (*api.JobSubmitResponse, error) {
+		switch owned, err := authorizeOwnership(ctx, request.Queue); {
```

Comment:
This is rather confusing, owned/err aren't referenced by any case statements. It would be better on its own line.

I think if/else is clearer here, not case.

Reply:
Maybe you are not familiar but in go you can define variables in switch statement. Most common place is when you do type switching like here: https://github.com/G-Research/armada/blob/1b933fe66485c8aba7112b9ab0ab4d147a0af336/pkg/api/event_util.go#L35

If you follow up with semicolon like here, it doesn't switch by any value. You can have switch statement in go without switch value:

```go
var i;
var j;

switch {
  case i == 0:
  case i == 1:
  case j == 3:
  case i%j == 0:
  default:
}

```

This is very useful to define code that needs more complex if/else statements. I hopes this makes sense.

Note: I've moved 
```go
owned, err := authorizeOwnership(ctx, request.Queue)
```
outside of switch statement.

Reply:
Yes I'm aware this is possible, but still think it's clearer to have `owned, err := authorizeOwnership(ctx, request.Queue)` outside, thanks for changing